File: backend/src/main/java/com/politeai/infrastructure/email/ConsoleEmailService.java
package com.politeai.infrastructure.email;

import com.politeai.domain.auth.service.EmailService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

@Service
@Profile("!prod & !resend")
@Slf4j
public class ConsoleEmailService implements EmailService {

    @Override
    public void sendVerificationEmail(String to, String code) {
        log.info("""
                ========================================
                [DEV] 이메일 인증코드 발송
                수신: {}
                인증코드: {}
                ========================================""", to, code);
    }
}

File: app/email/console_email_service.py
import logging

logger = logging.getLogger(__name__)


class ConsoleEmailService:
    def send_verification_email(self, to: str, code: str) -> None:
        logger.info(
            "\n========================================\n"
            "[DEV] 이메일 인증코드 발송\n"
            "수신: %s\n"
            "인증코드: %s\n"
            "========================================",
            to,
            code,
        )
